yarn-project/p2p/src/mem_pools/tx_pool/aztec_kv_tx_pool.ts

```diff
@@ -264,32 +272,35 @@ export class AztecKVTxPool implements TxPool {
   }
 
   /**
-   * Archive a tx for future reference.
+   * Archives a tx for future reference. The number of archived txs is limited by the specified archivedTxLimit.
    * @param tx - The transaction to archive.
    */
-  private archiveTx(tx: Tx) {
-    while (this.#archivedTxs.size() >= this.#archivedTxLimit) {
-      const tailIdx = this.#archivedTxTail.get() ?? 0;
-      const txHash = this.#archivedTxIndices.get(tailIdx);
-      if (txHash) {
-        void this.#archivedTxs.delete(txHash);
-        void this.#archivedTxIndices.delete(tailIdx);
+  private archiveTx(tx: Tx): Promise<void> {
+    return this.#archive.transaction(() => {
+      let headIdx = this.#archivedTxHead.get() ?? 0;
+      let tailIdx = this.#archivedTxTail.get() ?? 0;
+
+      while (headIdx - tailIdx >= this.#archivedTxLimit) {
+        const txHash = this.#archivedTxIndices.get(tailIdx);
+        if (txHash) {
+          void this.#archivedTxs.delete(txHash);
+          void this.#archivedTxIndices.delete(tailIdx);
+        }
+        void this.#archivedTxTail.set(++tailIdx);
```

Comment:
Let's avoid one write per loop iteration. We can set this to the final value at the end of the function.